pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -52,10 +52,10 @@ func (e *ErrorWithCodes) Error() string {
 }
 
 var (
-	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|InvalidAuthenticationTokenTenant|SignatureDoesNotMatch|Authentication failed|AuthFailure|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId)`)
+	unauthenticatedRegexp               = regexp.MustCompile(`(?i)(Unauthorized|InvalidAuthenticationTokenTenant|Authentication failed|AuthFailure)`)
```

Comment:
Mmh, if this is the name of the error code 401 then you are right matching the string `Unauthorized` to the error `Unauthenticated`. But this is about the error message. Really not sure in which bucket to place this. Maybe somebody else has experience with the error messages. I would probably put it in `unauthorizedRegexp `.

Reply:
That is what was confusing. 
@rfranzke WDYT?

Reply:
I guess `Unauthorized` should not be here since it indicates missing permissions, so 403, right?